haystack/components/evaluators/llm_evaluator.py

```diff
@@ -0,0 +1,230 @@
+import json
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+from haystack import component, default_from_dict, default_to_dict
+from haystack.components.builders import PromptBuilder
+from haystack.components.generators import OpenAIGenerator
+from haystack.utils import Secret, deserialize_secrets_inplace
+
+
+@component
+class LLMEvaluator:
+    """
+    Uses an LLM to evaluate inputs based on a prompt containing instructions and examples.
+
+    The default API requires an OpenAI API key to be provided as an environment variable "OPENAI_API_KEY".
+    The inputs are lists that are user-defined depending on the desired metric.
+    The output is a dictionary with a key `results` containing a list of evaluation results.
+    Each result is a dictionary with user-defined keys and values of either 0 for FALSE or 1 for TRUE respectively.
+
+    Usage example:
+    ```python
+    from haystack.components.evaluators import LLMEvaluator
+    evaluator = LLMEvaluator(
+        instructions="Is this answer problematic for children?",
+        inputs=[("responses", List[str])],
+        outputs=["score"],
+    )
+    RESPONSES = [
+        "Football is the most popular sport with around 4 billion followers worldwide",
+        "Python language was created by Guido van Rossum.",
+    ]
+    results = evaluator.run(responses=RESPONSES)
+    ```
+    """
+
+    def __init__(
+        self,
+        instructions: str,
+        inputs: List[Tuple[str, Type]],
+        *,
+        outputs: Optional[List[str]] = None,
+        api: str = "openai",
+        api_key: Secret = Secret.from_env_var("OPENAI_API_KEY"),
+        examples: Optional[List[Dict[str, Any]]] = None,
```

Comment:
I wouldn't limit us to always providing examples. Instructions can also work without them, no?

Reply:
That's a fair point, and it also relates to the aforementioned question. 

IMO, this component should only be prescriptive in providing a general shape of the inputs (in the larger sense) that are required by the component to performs its function. Almost always, few-shot prompting is better than zero-shot prompting. So, we should push the consumer of this component to provide as much information as possible to facilitate a high quality response (particularly since we're talking about evaluation here).

Reply:
Okay, I made outputs and examples non-optional. If we want to make them optional at a later point it's not a breaking change so let's move fast. 👍